hoprd/rest-api/src/session.rs

```diff
@@ -43,6 +51,159 @@ lazy_static::lazy_static! {
     ).unwrap();
 }
 
+#[serde_as]
+#[derive(Debug, Clone, Serialize, Deserialize, utoipa::IntoParams, utoipa::ToSchema)]
+#[into_params(parameter_in = Query)]
+#[serde(rename_all = "camelCase")]
+pub(crate) struct SessionWebsocketClientQueryRequest {
+    #[schema(required = true)]
+    #[serde_as(as = "DisplayFromStr")]
+    pub destination: PeerId,
+    #[schema(required = true)]
+    pub hops: u8,
+    #[schema(required = true)]
+    #[serde_as(as = "Vec<DisplayFromStr>")]
+    pub capabilities: Vec<SessionCapability>,
+}
+
+impl SessionWebsocketClientQueryRequest {
+    pub(crate) fn into_protocol_session_config(self) -> Result<SessionClientConfig, HoprLibError> {
+        Ok(SessionClientConfig {
+            peer: self.destination,
+            path_options: RoutingOptions::Hops((self.hops as u32).try_into()?),
+            target_protocol: IpProtocol::TCP, // TODO: change this to a more reasonable value
+            target: "127.0.0.1:0"
```

Comment:
Must be configurable in `SessionWebsocketClientQueryRequest`

Reply:
Indeed it is in the newer version of the PR.

Reply:
But upon seeing this, I'm 99.9% sure I want this behavior changed in 3.0. This leaks too much knowledge and assumptions about the exit node setup to the entry node.